lib/seahorn/HornifyFunction.cc

```diff
@@ -227,6 +291,62 @@ void SmallHornifyFunction::runOnFunction(Function &F) {
   side.clear();
   s.reset();
 
+  // Generates rules for synthesis.
+  for (auto &BB : F) {
+    auto partials = getPartialFnsToSynth(BB);
```

Comment:
potential performance penalty when synthesis is not enabled

Reply:
Updated. The code only looks at blocks that call `sea.synth.assert`. The check is constant in the size of the module, and linear in the number of calls to `sea.synth.assert`.